[DO NOT MERGE] Documentation update review (#172)
canonical/kafka-operator: docs/how-to/h-deploy.md

juju show-controller | yq '.[].details.cloud'
```

> **IMPORTANT** If the cloud is `k8s`, please refer to the [Charmed Kafka K8s documentation](/t/charmed-kafka-k8s-documentation/10296) instead.

You can find more information on how to bootstrap and configure a controller for different 
clouds [here](https://juju.is/docs/juju/manage-controllers#heading--bootstrap-a-controller). 
Make sure you bootstrap a `machine` Juju controller. 

## Juju Model setup

You can create a new Juju model using 

```
juju add-model <model>
```

Alternatively, you can use a pre-existing Juju model and switch to it by 

```
juju switch <model-name>
```

Make sure that the model is **not** a `k8s` type. The type of the model 
can be obtained by 

```
juju show-model | yq '.[].type'
```

> **IMPORTANT** If the model is `k8s`, please refer to the [Charmed Kafka K8s documentation](https://discourse.charmhub.io/t/charmed-kafka-k8s-documentation/10296) instead.

## Deploy Charmed Kafka and Charmed ZooKeeper

The Kafka and ZooKeeper charms can both be deployed as follows:

```shell
$ juju deploy zookeeper --channel 3/edge -n <zookeeper-units>
$ juju deploy kafka --channel 3/edge -n <kafka-units>
```

> After this, it is necessary to connect them:

Reviewer: ```suggestion
After this, it is necessary to [integrate](https://juju.is/docs/juju/relation) them:
```

Author: yes, I'm afraid we may have a bit of "relate" all over the place right now (even in other docs that are not included in this PR). I'll try to do a search and replace, but in one commit to hopefully address this consistently

```shell

> $ juju relate kafka zookeeper

Reviewer: ```suggestion
$ juju integrate kafka zookeeper
```

> After this, it is necessary to connect them:
> ```shell
> $ juju relate kafka zookeeper
> ```

Reviewer: ```suggestion
After this, it is necessary to [integrate](https://juju.is/docs/juju/relation) them:
```shell
$ juju integrate kafka zookeeper
```
```

> We recommend values of `<kafka-units>` of 3 and `<zookeeper-units>` of 5, for 
ensuring reasonable high-availability margins.

Once all the units show as `active|idle` in the `juju status` output, the deployment 
should be ready to be used. 

## (Optional) Create an external admin users

After being deployed, the Kafka cluster is *secure by default* and, as a consequence, 
it won't expose any external listener. 
In fact, ports are only opened when client applications are related, also 

> depending on the protocols to be used. Please refer to [this table](TODO) for 

Author: yes, this will be updated once we start publishing this content into Discourse

more information about the available listeners and protocols. 

It is however generally useful for most of the use-cases to create a first admin user
to be used to manage the Kafka cluster (either internally or externally). 

To create an admin user, deploy the [Data Integrator Charm](https://charmhub.io/data-integrator) with 
`extra-user-roles` set to `admin`

```shell
juju deploy data-integrator --channel stable --config topic-name=test-topic --config extra-user-roles=admin
```

> and relate to the Kafka charm

Reviewer: ```suggestion
and integrate with the Kafka charm
```

```shell

> juju relate data-integrator kafka

Reviewer: ```suggestion
juju integrate data-integrator kafka
```

```

To retrieve authentication information such as the username, password, etc. use

```shell

> juju run data-integrator/leader get-credentials

Author: it should! if there is one unit only, that's obviously the leader
